addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/TestRemoteFileInclude.java

```diff
@@ -40,23 +42,40 @@
 
     /** the various prefixes to try, for each of the remote file targets below */
     private static final String[] REMOTE_FILE_TARGET_PREFIXES = {
-        "http://", "", "HTTP://", "https://", "HTTPS://", "HtTp://", "HtTpS://"
+        "http://",
+        "",
+        "HTTP://",
+        "https://",
+        "HTTPS://",
+        "HtTp://",
+        "HtTpS://",
+        // Null Byte payload, incase validator for the remote file inclusion is
+        // vulnerable to null byte ie say validator is written in C/C++ where Null
+        // character marks end of string then that validator will not read characters
+        // after null byte hence validation can be bypassed.
```

Comment:
didn't got your point fully. but following is my observation if null byte is included within the url like http://%00google.com or http://user:%00@google.com it marks the url as invalid but if null byte is added before the http scheme, it doesn't mark it invalid. This behavior i had tested in DVWA application.

Reply:
I'm saying that the URL actually used would be e.g. `\0http://google.com` which would not have a valid scheme (i.e. `\0http`), of course it could be trimmed by the target before use. IMO these assumptions should be mentioned in the comment.

Did you try with just the null byte payloads when testing with DVWA? The non-null byte payloads already triggers the issue (with null bytes it has no URL to access `[:error] [pid 580] [client 172.17.0.1:44742] PHP Warning:  include(): Failed opening '' for inclusion (include_path='.:/usr/share/php:../../external/phpids/0.6/lib/') in /var/www/html/vulnerabilities/fi/index.php on line 36`).

Reply:
let me try again but i remember i tried %00http://google.com did you also tried same ?

Reply:
Yes, that (and the others) gives the above warning (and the final page does not have any additional content).

Reply:
oh ok... let me try again, if fails then should i add a comment of the assumption that if trimmed then it will work ?

Reply:
Yeah.

Reply:
yes, it is not working, i think some thing wrong i did that day. will update the comment with assumption.

Reply:
@thc202 , i have included assumption.

thanks,
Karan